src/Pages/ProductDetails/Components/GuestReview/GuestReview.js

```diff
@@ -0,0 +1,141 @@
+import React, { Component } from 'react';
+import './GuestReview.scss';
+
+class GuestReview extends Component {
+  state = {
+    text: '',
+    guestReviews: [],
+    num: 1,
+    id: 0,
+  };
+
+  handleInput = e => {
+    this.setState({
+      text: e.target.value,
+    });
+  };
+
+  getToday() {
+    const date = new Date();
+    const year = date.getFullYear();
+    const month = ('0' + (1 + date.getMonth())).slice(-2);
+    const day = ('0' + date.getDate()).slice(-2);
+    return year + '-' + month + '-' + day;
+  }
+
+  handleKeyPress = e => {
+    if (e.key === 'Enter') {
+      if (!this.state.text) {
+        e.preventDefault();
+      } else {
+        return this.addComment();
+      }
+    }
+  };
```

Comment:
```suggestion
  addComment = e => {
    if (e.key !== 'Enter') return;
    if (!this.state.text) return;
    
    const { id, text, guestReviews, reviewer } = this.state;
    const date = this.getToday();
    const num = guestReviews.length;

    this.setState({
      guestReviews: [
        ...guestReviews,
        {
          text: text,
          id: id + 1,
          num: guestReviews.length + 1,
          reviewer: 'dongdong2',
          date: date,
        },
      ],
      text: '',
    });
  };
```
- 함수가 실행되어야 하는 예외 상황을 먼저 앞에서 걷어내주고 실질적인 실행 로직을 뒷쪽에 배치해주면 다음과 같이 if문의 중괄호 깊이가 깊어지지 않도록 코드를 작성할 수 있습니다.
- 결국 이 함수가 하는 일이 댓글 작성하는 것 같은데요. 이 함수의 이름을 addComment로 바꾸고 기존에 addComment에 있던 로직을 이 힘수로 옮겨주는건 어떨까 싶네요!